src/serializer/Referentializer.js

```diff
@@ -194,8 +194,11 @@ export class Referentializer {
 
           // Already referentialized in prior scope
           if (residualBinding.declarativeEnvironmentRecord) {
-            invariant(residualBinding.scope);
-            instance.scopeInstances.add(residualBinding.scope);
+            const { scope } = residualBinding;
```

Comment:
Is there a reason for the switch to the pattern-matching? Seems like overkill in this case